src/genericSchemeRegistry/schemes/DaoGenericSchemesForms.json

```diff
@@ -0,0 +1,31 @@
+{
+  "daos": [
+    {
+      "name": "DutchX",
```

Comment:
Ok, I am not a fan of this change.

So the purpose of tis form generation stuff is that is (1) to make it as easy as possible to add scheme interfaces, ideally without writing code (2) make it modular: it should be easy to add and remove particular interfaces, and (in the far (?) future) bundle them separately 

The "one-file-per-genericScheme" format is definitly simpler: to add a new interface, you can just take an existint json file as a model, adapt it, add it, and you are done. The current format makes this much harder.

It is also more modular, in a sort of trivial way: to add or remove an interface, just add or remove the json file.

The only point where this new scheme is superior (afaics) is that it provides a mechanism for re-using code. But I personally do not think this is worth the price.

Reply:
> The "one-file-per-genericScheme" format is definitly simpler: to add a new interface, you can just take an existint json file as a model, adapt it, add it, and you are done. The current format makes this much harder.

In the separate-file pattern, when adding a new DAO (we should be talking about DAOs here)  one would have to also add a new `import` statement and a reference to it the import in the typescript.  The only way around this would be some dynamic importing, but then you move the problem to the webpack config to know how to bundle the json files.

In contrast, with I am proposing, adding a new DAO involves only a copy/paste/adapt of a simple section of JSON.  Period.  Even without noting the additional work regarding imports (in the para above), this would still be no more complicated than having separate files.

Reply:
> and (in the far (?) future) bundle them separately

I think in any future case, whatever query we run should ultimately (in the future beyond the current hard-coded local json, which is a hack) be running against a single data store, ideally ethereum or ipfs or whatever.  A single json file follows the pattern of a single data store and is easier to code against (no fluctuating imports, see the comment above).

The internal architecture of how the data is arranged within the store should not be the concern of the application typescript. The import pattern violates this separation of concerns.

Reply:
> We should be talking about DAOs here

Why? I think we should be talking aobut Schemes here, not DAOs.

> have to also add a new `import` statement

Dynamic loading is the way, and the JSON files would not be part of the bundle (see my remark about 1000s of custom components), but loaded separately.

> simple section of JSON

well, that is exactly my point,  that it is a section, so different DAOs with different schemes would ahve to "edit" the **same** file. 

> I think that in the future...

So clearly we do not have the same ideas about how this software would evolve and become more modular. In my view it would not be a single application, but an application framework with pluggable parts (of which these JSON files are an embrionic form)  - and so having a single file is quite the opposite of where I see this going. 

But it is way too early anyway to say something definitive about that.

The bottom line is: we have an existing, working, non-broken, implementation, this PR makes that implementation more complex,  so I'd prefer to not merge this...

Reply:
>We should be talking about DAOs here
>> Why? I think we should be talking aobut Schemes here, not DAOs.

For example, if dxDAO wanted the dao-to-dao, they can't as the JSON is currently structured (am I wrong?).  This scenario involves two different schemes in one DAO.  I think that we want that a DAO can customize what schemes it wants to use and how it wants to use them (the set of actions/forms).

Reply:
> so different DAOs with different schemes would ahve to "edit" the same file

Where, when and how are they going to do that?

Reply:
>  single application, but an application framework with pluggable parts 

Indeed soon, and I would note that "pluggable parts", while it has merits in some scenarios, is not an automated process from the DAO's perspective.  With specific respect to GenericScheme I envision a process that _can_ be fully automated, where resorting to a plugin would be a last resort when generic functionality will not suffice.

Reply:
Correcting myself:  It appears I _was_ wrong about:

> For example, if dxDAO wanted the dao-to-dao, they can't as the JSON is currently structured (am I wrong?).

Why?  Simply by virtue of the GenericScheme being registered twice, each with a different target contract address, the Schemes list page will ,for each, find the JSON containing the target contract address.